sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/DetailAstRootHolder.java

```diff
@@ -0,0 +1,66 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+
+/**
+ * Holds the link to the root of AST.
+ * Copy of FileContentsHolder.
+ * @author Mike McMahon
+ * @author Rick Giles
+ * @author attatrol
+ * @see com.puppycrawl.tools.checkstyle.checks.FileContentsHolder
+ */
+public class DetailAstRootHolder extends AbstractCheck {
```

Comment:
@romani We are trying to remove holders, but I don't think there is anyway around doing this here.

Reply:
@michaldo Please remove this class and all instances of it.